core/modules/user/store/actions.ts

```diff
@@ -49,7 +50,9 @@ const actions: ActionTree<UserState, RootState> = {
    * Login user and return user profile and current token
    */
   async login ({ commit, dispatch }, { username, password }) {
-    const resp = await UserService.login(username, password)
+    const userAfterHook = userHooksExecutors.beforeUserAuthorize({ username, password })
+    const resp = await UserService.login(userAfterHook.username, userAfterHook.password)
```

Comment:
hmm... with this you are not able to connect with the external login provider because you still use `UserService.login` which calls our API. So maybe this hook should return response and if it's null then you can use `UserService.login` instead.

Eg. you can create dedicated action for that:
```javascript
async attemptLogin (context, { login, password }) {
 const resp = await userHooksExecutors.beforeUserAuthorize({ username, password })

 if (resp) {
  return resp
 }

 return UserService.login(username, password)
}
```
and in the login action:
```javascript
const resp = await dispatch('attemptLogin', { username, password })
```